pennylane/gradients/parameter_shift.py

```diff
@@ -174,19 +215,16 @@ def expval_param_shift(tape, argnum=None, shifts=None, gradient_recipes=None, f0
     argnum = argnum or tape.trainable_params
 
     gradient_tapes = []
-    gradient_coeffs = []
-    shapes = []
-    unshifted_coeffs = []
-
-    fns = []
+    # Each entry for gradient_data will be a tuple with content
+    # (num_tapes, coeffs, fn, unshifted_coeff)
+    gradient_data = []

```

Comment:
This is the new data format. It saves verbosity because it bunches four properties together that have low complexity but were taking up many lines of code.